src/node/stage/approved.rs

```diff
@@ -1935,6 +2377,21 @@ impl Approved {
         }
     }
 
+    #[cfg(feature = "mock_base")]
+    // Returns whether node has completed the full joining process
+    pub fn is_ready(&self, core: &Core) -> bool {
+        if self
+            .shared_state
+            .our_members
+            .elder_candidates(core.network_params.elder_size)
+            .contains_key(core.id().name())
+        {
+            self.is_our_elder(core.id())
+        } else {
+            true
+        }
+    }
```

Comment:
Why do we need this?

Reply:
the exsiting testing code only checks `joined` for the new node and will bootstrap next immediately.
doing so, some new node could be bootstrapped too early which messed up with its own DKG voting round.
This is to ensure that every supposed to have DKG node does completed, then to bootstrap next.

Non elder node shall not be affected.

Reply:
OK, I see. So given the current ongoing discussion about reducing the use of mocks, I think it would be much better to not do this and instead modify the tests themselves. For example, I recently added `poll_until_minimal_elder_count` which waits until the section reaches `MIN_ELDER_SIZE` elders before proceeding. Perhaps that, or something similar to that could be used in this case too?

Reply:
ok, I will have a check if can use `poll_until_minimal_elder_count ` instead.

Reply:
this is actually being used by the `node_joined` testing function call, which inside doesn't know whether the targeted node will be an elder or not.
refactoring this will mean modify many calling places within the testing code. 
I'd prefer to do it in a separate PR.

Reply:
OK, I see. I'm just worried this might potentially hide some issues. For example, there is a test `simultaneous_joining_nodes` which tries to test that the network handles many nodes joining at the same time. With this change, we are basically implicitly limiting the number of simultaneous nodes which makes the test less aggressive and thus less likely to catch bugs. There might be other tests similarly affected by this.

That said, I'm OK with postponing this to a separate PR. Just maybe add a `TODO` comment there to remind us about this.

Reply:
Adding multiple adult nodes is not affected. However, I agree with the intial phase that all new nodes are elder, this part could be affected.
I will add a TODO comment in.